drivers/regulator/regulator_nxp_vref.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2023 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_vref
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/dt-bindings/regulator/nxp_vref.h>
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/linear_range.h>
+#include <soc.h>
```

Comment:
please include HAL headers, avoid soc.h

Reply:
why? what HAL headers? I'm not using a HAL, I'm using SOC definitions

Reply:
@gmarull is something like this what you prefer instead of `#include <soc.h>`?
```
#ifdef CONFIG_SOC_LPC5536
#include "lpc5536.h"
#elif CONFIG_OTHER_SOC
#include "other_soc.h"
#elif ...
....
#else 
#error "SOC doesnt have VREF driver"
#endif
```